pkg/storage/read_as_of_iterator.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package storage
+
+import "github.com/cockroachdb/cockroach/pkg/util/hlc"
+
+// ReadAsOfIterator wraps an iterator that implements the SimpleMVCCIterator
+// such that any valid MVCC key returned will have a timestamp less than the
+// asOf timestamp, is neither a delete tombstone, nor shadowed by a delete
+// tombstone less than the asOf timestamp.
+type ReadAsOfIterator struct {
```

Comment:
Good point. The iterator does not need to handle write intents, since it's currently only used by restore, which does not encounter any write intents.

Reply:
Cool, let's add a mention of that in this comment then.